source/MaterialXFormat/XmlIo.cpp

```diff
@@ -120,36 +117,35 @@ void elementToXml(ConstElementPtr elem, xml_node& xmlNode, const XmlWriteOptions
     }
 }
 
-void xmlDocumentFromFile(xml_document& xmlDoc, string filename, const string& searchPath)
+void xmlDocumentFromFile(xml_document& xmlDoc, FilePath filename, FileSearchPath searchPath)
 {
-    FileSearchPath fileSearchPath = FileSearchPath(searchPath);
-    fileSearchPath.append(getEnvironmentPath());
+    searchPath.append(getEnvironmentPath());
```

Comment:
I believe append() does not check to see if it already exists so the side-effect is that calling this function multiple times with the same input searchPath could keep on adding the same environment path.
Seems this method should continue to create a local variable and append to that instead.

Reply:
Although we could be more clear about this, the searchPath variable is _already_ a local copy, because it's now passed by value instead of by reference.

Reply:
Ah. I missed this. I was assuming it was still by reference. Please ignore the comment then.